cidata.py

```diff
@@ -0,0 +1,431 @@
+                                       `10.0.0.1`.
+                          type: string
+                          required: true
+        config_file:
+            description: path to the config.json file
+            type: string
+            required: false
+        userdata:
+            description: the cloud-init user data payload to pass into the virtual machine
+            type: string
+            required: false
+    return:
+        description: |
+            A tuple with a boolean flag stating the build was successful or not and
+            the output or error message.
+        type: tuple
+    """
+    # Define message
+    messages = {
+        1000: f'1000: Successfully created {domain_path}/metadata and {domain_path}/userdata on both PodNet nodes.',
+        2111: f'2011: Config file {config_file} loaded.',
+        3011: f'3011: Failed to load config file {config_file}, It does not exist.',
+        3012: f'3012: Failed to get `ipv6_subnet` from config file {config_file}',
+        3013: f'3013: Invalid value for `ipv6_subnet` from config file {config_file}',
+        3014: f'3014: Failed to get `podnet_a_enabled` from config file {config_file}',
+        3015: f'3015: Failed to get `podnet_b_enabled` from config file {config_file}',
+        3016: f'3016: Invalid values for `podnet_a_enabled` and `podnet_b_enabled`, both are True',
+        3017: f'3017: Invalid values for `podnet_a_enabled` and `podnet_b_enabled`, both are False',
+        3018: f'3018: Invalid values for `podnet_a_enabled` and `podnet_b_enabled`, one or both are non booleans',
+        3021: f'3021: Failed to connect to the enabled PodNet from the config file {config_file} for create_metadata payload',
+        3022: f'3022: Failed to create metadata file {domain_path}/metadata on the enabled PodNet. Payload exited with status ',
+        3023: f'3023: Failed to connect to the enabled PodNet from the config file {config_file} for create_userdata payload',
+        3023: f'3024: Failed to create userdata file {domain_path}/userdata on the enabled PodNet Payload exited with status ',
+        3031: f'3031: Successfully created `metadata` and `userdata` in {domain_path}/ on enabled PodNet but failed to connect to the disabled PodNet '
+              f'from the config file {config_file}',
+        3032: f'3032: Successfully created `metadata` and `userdata` in {domain_path}/ on enabled PodNet but failed to create on the disabled PodNet. '
+               'Payload exited with status ',
+        3033: f'3033: Successfully created `metadata` and `userdata` in {domain_path}/ on enabled PodNet but failed to connect to the disabled PodNet. '
+              f'from the config file {config_file}',
+        3034: f'3034: Successfully created `metadata` and `userdata` in {domain_path}/ on enabled PodNet but failed to create on the disabled PodNet. '
+               'Payload exited with status ',
+    }
+
+    metadata_json = json.dumps(
+        metadata,
```

Comment:
The `metadata` is not passed in the request. A jinja template is required (see nextcloud) and it would be constructed based on the contract.

Reply:
We talked about changing the possibility of changing the contract yesterday. So is that no longer possible?

Reply:
Also, should we really use a template when the net result is going to be a serialized data structure (the `metadata` file is in JSON format) anyway? This strikes me as needlessly error prone.